ws-services/src/main/java/org/egov/waterconnection/service/MeterInfoValidator.java

```diff
@@ -21,40 +21,71 @@
-						|| waterConnectionRequest.getWaterConnection().getMeterInstallationDate() == 0) {
-					errorMap.put("INVALID_METER_INSTALLATION_DATE",
-							"Meter Installation date cannot be null or negative");
-				}
-				HashMap<String, Object> addDetail = mapper.convertValue(
-						waterConnectionRequest.getWaterConnection().getAdditionalDetails(), HashMap.class);
-				if (StringUtils.isEmpty(addDetail)
-						|| addDetail.getOrDefault(WCConstants.INITIAL_METER_READING_CONST, null) == null) {
-					errorMap.put("INVALID_INITIAL_METER_READING", "Initial meter reading can not be null");
-				} else {
-					BigDecimal initialMeterReading = new BigDecimal(
-							String.valueOf(addDetail.get(WCConstants.INITIAL_METER_READING_CONST)));
-					if (initialMeterReading.compareTo(BigDecimal.ZERO) == 0) {
-						errorMap.put("INVALID_INITIAL_METER_READING", "Initial meter reading can not be zero");
-					}
-				}
+			if (WCConstants.METERED_CONNECTION
+					.equalsIgnoreCase(waterConnectionRequest.getWaterConnection().getConnectionType())) {
+				validateMeteredConnectionRequst(waterConnectionRequest, errorMap);
+			}
+		}
+	}
+	
+	private void handleModifyConnectionRequest(WaterConnectionRequest waterConnectionRequest,
+			Map<String, String> errorMap) {
+		if (WCConstants.APPROVE_CONNECTION
+				.equalsIgnoreCase(waterConnectionRequest.getWaterConnection().getProcessInstance().getAction())) {
+			if (WCConstants.METERED_CONNECTION
+					.equalsIgnoreCase(waterConnectionRequest.getWaterConnection().getConnectionType())) {
+				validateMeteredConnectionRequst(waterConnectionRequest, errorMap);
+			}
+		}
+	}
+	
+	@SuppressWarnings("unchecked")
+	private void validateMeteredConnectionRequst(WaterConnectionRequest waterConnectionRequest,
+			Map<String, String> errorMap) {
 
+		if (waterConnectionRequest.getWaterConnection().getMeterId() == null) {
+			errorMap.put("INVALID_METER_ID", "Meter Id cannot be empty");
+		}
+		if (waterConnectionRequest.getWaterConnection().getMeterInstallationDate() == null
+				|| waterConnectionRequest.getWaterConnection().getMeterInstallationDate() < 0
+				|| waterConnectionRequest.getWaterConnection().getMeterInstallationDate() == 0) {
+			errorMap.put("INVALID_METER_INSTALLATION_DATE",
+					"Meter Installation date cannot be null or negative");
+		}
+		HashMap<String, Object> addDetail = mapper.convertValue(
+				waterConnectionRequest.getWaterConnection().getAdditionalDetails(), HashMap.class);
+		if (StringUtils.isEmpty(addDetail)
+				|| addDetail.getOrDefault(WCConstants.INITIAL_METER_READING_CONST, null) == null) {
+			errorMap.put("INVALID_INITIAL_METER_READING", "Initial meter reading can not be null");
+		} else {
+			BigDecimal initialMeterReading = BigDecimal.ZERO;
+			initialMeterReading = new BigDecimal(
+					String.valueOf(addDetail.get(WCConstants.INITIAL_METER_READING_CONST)));
+			if (initialMeterReading.compareTo(BigDecimal.ZERO) == 0) {
```

Comment:
Why `<=0` is not being compared here? <=0 should be a invalid meter reading, this check currently may allow -ve numbers

Reply:
Actually, negative number validation is handled at UI